airbyte-integrations/connectors/destination-databricks/src/main/kotlin/io/airbyte/integrations/destination/sync/SyncOperations.kt

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.sync
+
+import io.airbyte.cdk.integrations.destination.StreamSyncSummary
+import io.airbyte.protocol.models.v0.StreamDescriptor
+
+/** Connector sync operations */
+interface SyncOperations {
```

Comment:
checking my understanding: this basically lets us shim StreamOperations into the AsyncStreamConsumer?

Reply:
thats how it shaped, moved a lot of multi-stream state logic (or still moving) from DefaultTyperDeduper to shoehorn into existing AsyncSConsumer